tensorflow/tools/ci_build/osx/arm64/tensorflow_as_build_nightly.Jenkinsfile

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+pipeline {
+    agent none
+
+    stages {
+        stage("Build Tensorflow") {
+            parallel {
+                stage("Python 3.8") {
+                    agent {
+                        label "nightly-build"
+                    }
+                    environment {
+                        PYENV_ROOT="$HOME/.pyenv"
+                        PATH="$PYENV_ROOT/shims:/opt/homebrew/bin/:$PATH"
+
+                    }
+                    steps {
+                        dir('tensorflow') {
+
+                            sh '''
+                                pyenv init -
+                                pyenv global 3.8.13
+                            '''
+
+                            sh 'python --version'
+
+                            git branch: "master",
```

Comment:
I assume this is the case for all of the pip and nonpip tests as well?

Reply:
Actually, I think we might not need the pip tests anymore as we are in the process of disabling it internally. @MichaelHudgins Can you confirm if these can be disabled in the Apple Silicon CI as well?

Assuming we disable the pip tests, I think the CI should look like this:
* Continuous - Runs from the master branch and runs the nonpip tests
* Nightly - Runs from the Nightly branch. 2 separate jobs - wheel building and nonpip tests
* GitHub presubmits for PRs to Master - We don't have this right now and I don't think we need immediately but it is something we should look to enable.

The reason we want the nightly wheel building job to run from "nightly" is so that all the nightly wheels are built at the same commit. The nightly branch is synced to master sometime around 8pm everyday. Our internal nightly builds run sometime between 12AM to 3AM the next day so all the wheels are built at the same commit since the nightly branch is frozen until the next sync.  The nightly nonpip build then tells us how stable that nightly package is.

Reply:
Michael is OOO but I was able to get confirmation from rest of the team that we can disable pip tests in the Apple Silicon CI as well.